data/src/main/java/io/gnosis/kouban/data/managers/SearchManager.kt

```diff
@@ -78,8 +79,9 @@ data class TransactionTimestampFilter(
 ) : BoundaryFilter<Transaction>() {
 
     override fun apply(item: Transaction): Boolean {
-        return with(item.timestampAsDate()) {
-            (upperBound == null || before(upperBound)) && (lowerBound == null || after(lowerBound))
+        return with(item.timestampAsDate().onlyDate()) {
```

Comment:
Transaction.timestampAsDate - too much implementation details. Calling method .date would be enough

Reply:
I would prefer keeping it like that because Transactions have 3 different dates according to the swagger (modified, sumitted, execution) then we are going to have to specify to which date we refer to. Renaming to `date` provides no information on the context.

Reply:
what kind of date is used here? modified/submitted/execution? I would then rename the method to reflect it

Reply:
I don't know. That's the name received from the backend. Not sure if this data structure is finalised as it is.

Reply:
this is independent of the backend structure. Method name should semantically reflect its intended use, not give away implementation details

Reply:
I am not sure what you mean by that. I see this method the same way I see `asEthereumAddressString`

Reply:
Addtionally, I believe it's going to look weird if we had `Transaction.date().onlyDate()`

Reply:
here its probably the timestamp when block was collated, right @rmeissner ? So executionDate would probably be correct

Reply:
in this case its not an extension function, that converts address to string. Its a class method. And it should not reveal details of the class

Reply:
I am not sure if the client sure care what timestamp it is. Backend will only return 1 timestamp, until now I always found it confusing if multiple were returned. I would disregard what the transaction service does and only care about what the prototype api returns (and there it is just a timestamp)

Reply:
but it's a data class whose fields are all public. If you want I can change that to an extension function? I really don't understand the hold up. I am really sorry.

Reply:
For the naming: is this an extension on string? In that case I would call ot toDate() maybe ... No string preference

Reply:
It is just a method that multiplies the timestamp which is a long by 1000 and then returns it as a date. Thus, timestampAsDate.

Reply:
my point here: implementation details can change. Transaction could get date field and timestamp would be removed (exaggerated, but possible). Having name like timestampAsDate suggests that date is calculated based on timestamp. Another approach is by havign extension functions on timestamp itself: tx.timestamp.asDate()

Reply:
For me timestamp revers to a type here not the field. You could also call it epocheToDate. It is just to indicate that it is expected that the long is in seconds since 1.1.1970.